src/rsz/src/RepairSetup.cc

```diff
@@ -524,12 +524,13 @@ bool RepairSetup::swapPins(PathRef *drvr_path,
         }
 
         // Find the equivalent pins for a cell (simple implementation for now)
-        // stash them
-        if (equiv_pin_map_.find(cell) == equiv_pin_map_.end()) {
-            equivCellPins(cell, ports);
-            equiv_pin_map_.insert(cell, ports);
+        // stash them. Ports are unique to a cell so we can just cache by port
+        // and that should apply to all instances of that cell with this input_port.
+        if (equiv_pin_map_.find(input_port) == equiv_pin_map_.end()) {
+          equivCellPins(cell, input_port, ports);
+          equiv_pin_map_.insert(input_port, ports);
         }
-        ports = equiv_pin_map_[cell];
+        ports = equiv_pin_map_[input_port];
         if (ports.size() > 1) {
             resizer_->findSwapPinCandidate(input_port, drvr_port, load_cap,
                                            dcalc_ap, &swap_port);
```

Comment:
`ports.size() > 1` needs to become `!ports.empty()` with you change from keying on the input port rather than the cell.  Once you fix this please see if you actually need the metric updates that are in this PR.